airflow/providers/microsoft/mssql/hooks/mssql.py

```diff
@@ -63,11 +63,10 @@ def connection_extra_lower(self) -> dict:
     @property
     def sqlalchemy_scheme(self) -> str:
         """Sqlalchemy scheme either from constructor, connection extras or default."""
-        return (
-            self._sqlalchemy_scheme
-            or self.connection_extra_lower.get("sqlalchemy_scheme")
-            or self.DEFAULT_SQLALCHEMY_SCHEME
-        )
+        extra_scheme = self.connection_extra_lower.get("sqlalchemy_scheme")
+        if not self._sqlalchemy_scheme and extra_scheme and (":" in extra_scheme or "/" in extra_scheme):
+            raise RuntimeError("sqlalchemy_scheme in connection extra should not contain : or / characters")
+        return self._sqlalchemy_scheme or extra_scheme or self.DEFAULT_SQLALCHEMY_SCHEME
```

Comment:
What is the general precedence of considering connection parameters when there are multiple ways to set a parameter? 
I am under the impression that extras > constructor/hook params > default

Reply:
Generally, yes, and it also surprised me a bit. But I did not want to add even more `somewhat breaking` changes at the same time. One breaking thing at a time is breaking enough (even thought this one is technically a bugfix, not a breaking change).

But I can be convinced otherwise if others things it would be better to make it a breaking change as well and change the sequence.

Reply:
> If the user has set sqlalchemy_scheme via constructor but wants to override via extras, will we not allow it?

why does he need to override a very flexible value provided to the constructor by an extra provided in very less flexible way (connection).

If he wants to use the connection scheme he can skip providing a value to the  constructor, and if he want to override the connection scheme, he can provide the new value to the constructor.
We can easily create 10 instances of this hook, but we can easily create and manage 10 different connection. So for me it's perfect like that:
constructor/hook params > connection extras > default

Reply:
yes, not sure, how users use it. I saw it in a few other places where extras take precedence over constructor/hook params and hence was curious to understand. Changing the hook/constructor params would need DAG/code changes and the need to update that, whereas changing the extras would not need code changes and more or less like changing an environment variable. 
No strong opinion, just wanted to understand :)